### DISCUSSION TITLE
using SpringBootApplication

### DISCUSSION THREAD

**Discussion by wnleao** [sevdokimov/log-viewer#42] [Ideas]
Hi, I was taking a better look at the backend and I would like to know if you've considered using SpringBootApplication in the backend with its runner and @Controller(s) to handle the requests. But you know, there might be something that I may be overlooking.

**Comment by sevdokimov** [comment]
Log viewer can be embedded into another Java Web application as a library, so it must have as few dependencies as possible to avoid conflicts. And it should be as light as possible.
Spring rest controllers are nice, but LogViewer has 9 endpoints only. I think adding the dependency to spring boot doesn't worth it. Let's keep it as is.

I look forward to when you look at the integration tests (selenium), I feel my test implementation is ugly and may be improved)

**Reply by wnleao** [reply]
Hey! Okay, I understand that you want to get things as clean and light as possible, but I think that you've reimplemented some things that are given in spring and I believe would not increase your dependencies by much. No problem, I'm going to take a look at the integration tests.
